### DISCUSSION TITLE
Custom Theme

### DISCUSSION THREAD

**Discussion by krau5** [plouc/nivo#1841] [Q&A] [answered]
I haven't found a similar question or some guide about customizing a theme.
I mean we have the next object
```
const theme = { ... }
```
That has a type of `Theme` from `@nivo/core`. And it's like hard-coded, I do not see any opportunity to customize the theme and add some custom keys like I need.
Can I actually do that or it's not possible yet?

**Comment by krau5** [comment]
Say me, if I missed something, I'd add the info that's needed

**Comment by ys-mn** [comment]
Hello. I think you can use useTheme hook and console.log it. Also i was able to give other css properties to its objects and they were ok.

**Comment by plouc** [comment, marked as answer]
The theme object can be customized, using TypeScript will probably help for discoverability, but otherwise, you can have a look at the default theme here: https://github.com/plouc/nivo/blob/master/packages/core/src/theming/defaultTheme.js.

@krau5 it's not really hardcoded, it acts as a base for custom themes so that you don't have to define everything, your custom theme is merged with this one.

You can also have a look at the TS definition for it to see what you can customize: https://github.com/plouc/nivo/blob/master/packages/core/index.d.ts#L53.

I added a guide on how to use the theme object, but it's not complete yet: https://nivo.rocks/guides/theming/.
